ocs_ci/utility/kms.py

```diff
@@ -118,6 +119,73 @@ def gather_init_vault_conf(self):
             self.vault_tls_server = self.vault_conf["VAULT_TLS_SERVER_NAME"]
         self.vault_root_token = self.vault_conf["VAULT_ROOT_TOKEN"]
 
+    def vault_create_namespace(self):
+        """
+        Create a vault namespace if it doesn't exists
+
+        """
+
+        self.vault_namespace = get_default_if_keyval_empty(
+            config.ENV_DATA, "VAULT_NAMESPACE", constants.VAULT_DEFAULT_NAMESPACE
+        )
+
+        if (
+            not self.vault_namespace_exists(self.vault_namespace)
+            or self.vault_namespace == ""
+        ):
+            self.vault_namespace = (
+                f"{constants.VAULT_DEFAULT_NAMESPACE_PREFIX}-"
+                f"{get_running_cluster_id()}"
```

Comment:
I think we should go with cluster name instead. In the jenkins production jobs we have always new cluster name so we should be OK, if someone using cluster with the same name and he didn't cleanup properly previous deployment then at least user will know that they are doing something wrong if the namespace exists.  WDYT?  Without cluster name it will be hard for us to identify to which cluster the namespace belongs to. Also it will make the aws cleanup script easier to delete it with this pattern of the cluster name.

Reply:
it can be done in follow up pr I think, so approved this PR